src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+ * an optional control such as a slider or a number spinner that lets the user interact with the value in the socket.
+ * <p>
+ * This class both subscribes to and sends {@link SocketChangedEvent}s on the event bus, and acts as an intermediary
+ * between EventBus events and a JavaFX property.
+ */
+public class SocketControlView extends AnchorPane implements Initializable {
+    @FXML
+    private Label identifier;
+
+    @FXML
+    private StackPane controlPane;
+
+    private final EventBus eventBus;
+    private final Socket socket;
+    private final Property valueProperty;
+
+    public SocketControlView(EventBus eventBus, Socket<?> socket) {
+        checkNotNull(eventBus);
+        checkNotNull(socket);
+
+        this.eventBus = eventBus;
+        this.socket = socket;
+        this.valueProperty = new SimpleObjectProperty();
+
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("SocketControl.fxml"));
+            fxmlLoader.setRoot(this);
+            fxmlLoader.setController(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Always set the socket to the value of valueProperty, which may be bound to a GUI property.
+        this.valueProperty.addListener((observableValue, o, t1) -> this.socket.setValue(observableValue.getValue()));
+
+        this.eventBus.register(this);
+    }
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        SocketHint<?> socketHint = this.socket.getSocketHint();
+        Object[] domain = socketHint.getDomain();
+
+        // Set the label on the control based on the identifier from the socket hint
+        this.identifier.setText(socketHint.getIdentifier());
+
+        // Set an input control of some sort based on the "view" field of the socket hint
+        switch (socketHint.getView()) {
+            case NONE:
+                // The hint says not to show a control, so there's nothing to do here.
+                break;
+
+            case SPINNER: {
+                // To use a spinner, the socket should be able to hold a number
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType()));
+
+                SpinnerValueFactory spinnerValueFactory;
+                Number initialValue = (Number) socket.getValue();
+
+                Number min = Double.MIN_VALUE, max = Double.MAX_VALUE;
+                if (domain != null && domain.length == 2) {
+                    // A spinner can have a min and max, if the domain holds two values.
+                    min = (Number) domain[0];
+                    max = (Number) domain[1];
+                }
+
+                spinnerValueFactory = new SpinnerValueFactory.DoubleSpinnerValueFactory(
+                        min.doubleValue(), max.doubleValue(), initialValue.doubleValue());
```

Comment:
Look at line 88. The type declaration is on a different line. I'm wondering why you chose to do that.